tests/metrics/map_mem.sh

```diff
@@ -0,0 +1,351 @@
+	echo -e "\t\t\"shared\": $shared_avg," >> $JSON_FILE
+	echo -e "\t\t\"private\": $private_avg" >> $JSON_FILE
+}
+
+# it creates JSON file
+function create_json()
+{
+	local process_name="$1"
+	local samples_file="$2"
+
+	local mapping=""
+	local pss_mem=0
+	local shared_mem=0
+	local private_mem=0
+	local count=0
+
+	# accumulators
+	local shared_sum=0
+	local private_sum=0
+	local pssmem_sum=0
+
+	declare -A memory=(
+		["mapping"]=1
+		["pssmem"]=2
+		["shared"]=3
+		["private"]=4)
+
+	JSON_FILE="${process_name}.json"
+	output="$(cat "$samples_file" | sort)"
+
+	# start JSON file creation
+	echo "{" > "$JSON_FILE"
+
+	while read line;do
+		mapping=$(echo "$line" | awk -F "," -v column=${memory["mapping"]} '{print $column}')
+		pss_mem=$(echo "$line" | awk -F "," -v column=${memory["pssmem"]} '{print $column}')
+		shared_mem=$(echo "$line" | awk -F "," -v column=${memory["shared"]} '{print $column}')
+		private_mem=$(echo "$line" | awk -F "," -v column=${memory["private"]} '{print $column}')
+		count=$(( $count + 1 ))
+
+		# last state
+		if [ -z "$last_mapping" ];then
+			last_mapping="$mapping"
+		fi
+
+		# dump data
+		if [ "$mapping" != "$last_mapping" ];then
+			echo -e "\t\"$last_mapping\": {" >> $JSON_FILE
+			dump_avg_JSON $pssmem_sum $shared_sum $private_sum $count
+			echo -e "\t}," >> $JSON_FILE
+			shared_sum=0
+			private_sum=0
+			pssmem_sum=0
+			count=0
+			last_mapping="$mapping"
+		fi
+
+		# accumulate data
+		shared_sum=$(( $shared_sum + $shared_mem ))
+		private_sum=$(( $private_sum + $private_mem ))
+		pssmem_sum=$(( $pssmem_sum + $pss_mem ))
+
+	done <<< $(echo -e "$output")
+
+	# dump buffers
+	echo -e "\t\"$last_mapping\": {" >> $JSON_FILE
+	dump_avg_JSON $pssmem_sum $shared_sum $private_sum $count
+	echo -e "\t}" >> $JSON_FILE
+
+	# Finish JSON file creation
+	echo "}" >> $JSON_FILE
+}
+
+# take one sample of mapping
+function take_map_sample()
+{
+	local ps_name="$1"
+
+	# pmap tool shows a lot of
+	# data fields, this number to ignore
+	# head lines
+	local format_columns=23
+
+	# mapping tracking
+	local last_mapping=""
+	local mapping=""
+
+	# memory values
+	local pssmem=0
+	local pssmem_sum=0
+	local shared_memory=0
+	local shared_clean=0
+	local shared_dirty=0
+	local shared_hugep=0
+	local private_clean=0
+	local private_dirty=0
+	local private_hugep=0
+	local private_memory=0
+
+	# accumulators
+	local shared_sum=0
+	local private_sum=0
+	local anon_pssmem=0
+	local anon_shared=0
+	local anon_private=0
+
+	# Types of memory
+	#
+	# this array allows to filter:
+	# - PSS memory
+	# - (clean, dirty and hugetlb) shared memory
+	# - (clean, dirty and hugetlb) private memory
+	# form pmap tool output per mapping.
+	declare -A memory=(
+		["pssmem"]=8
+		["sclean"]=9
+		["sdirty"]=10
+		["shugep"]=17
+		["pclean"]=11
+		["pdirty"]=12
+		["phugep"]=18)
+
+	# getting pid of the process
+	pid=$(pidof $ps_name | awk '{print $1}')
+	if [ -z "$pid" ];then
+		docker rm -f  "$CC_ID"
+		die "process does not exist"
```

Comment:
Yes, sorry, I meant `$ps_name`, not `$pid` in this error message. Not a blocker though ;)